### PR TITLE
Upgrade Jenkins Server Example page

### PR DESCRIPTION
**Pull request by tgelpi-gmail**
Updated the steps to build Jenkins server based  upon current system level settings.

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

# Checklist:

- [ ] I have mentioned the appropriate type(scope), as referenced [here](https://jenkins-x.io/community/code/#the-commit-message) in the commit message and PR title for the semantic checks to pass.
- [ ] I have signed off the commit, as per instructions mentioned [here](https://jenkins-x.io/community/code/#how-to-sign-your-commits).
- [ ] Any dependent changes have already been merged.

### CODE REVIEW

File: content/en/v3/admin/guides/jenkins/example.md [hunk 1]
```diff
@@ -77,39 +80,48 @@ controller:
...
 ```
+# These components will merge with default versionStream/charts/jenkinsci/jenkins/values.yaml.gotmpl
 controller:
   installPlugins:
-    - kubernetes:1.29.0
-    - workflow-aggregator:2.6
-    - git:4.6.0
-    - configuration-as-code:1.47
-    - nodejs:1.3.11
+    - kubernetes:3546.v6103d89542d6
```
Comment (ankitm123) [inline content/en/v3/admin/guides/jenkins/example.md:88]: Is this version correct? They dont follow semver?

File: content/en/v3/admin/guides/jenkins/example.md [hunk 2]
```diff
@@ -77,39 +80,48 @@ controller:
...
+    - workflow-aggregator:2.7
+    - git:4.11.2
+    - configuration-as-code:1414.v878271fc496f
```
Comment (ankitm123) [inline content/en/v3/admin/guides/jenkins/example.md:91]: same comment, see above

File: content/en/v3/admin/guides/jenkins/example.md [hunk 3]
```diff
@@ -77,39 +80,48 @@ controller:
...
+Modification to the `$JX3HOME/jx3-gke-gsm/helmfiles/jx-jenkins/helmfiles.yaml` file is done to include the additional `values.yaml.gotmpl`.
 ```yaml
  - chart: jenkinsci/jenkins
    name: jenkins
    namespace: jx-jenkins
    values:
    - values.yaml
    - ../../versionStream/charts/jenkinsci/jenkins/values.yaml.gotmpl
-   - customPlugs.yaml
+   - values.yaml.gotmpl
```
Comment (ankitm123) [inline content/en/v3/admin/guides/jenkins/example.md:124]: Why this change? Is there some template requirement? or can we use a plain yaml?

**Review by ankitm123 — COMMENTED** [review]
minor comments ...

File: content/en/v3/admin/guides/jenkins/example.md (hunk 1 above)
Comment (tgelpi-gmail) [reply content/en/v3/admin/guides/jenkins/example.md:88]: The version was determined after logging into Jenkins app and viewing the manage plugins function. It is a later version of kubernetes plugin that is suitable for the version of Jenkins running (2.332.3). I

File: content/en/v3/admin/guides/jenkins/example.md (hunk 3 above)
Comment (tgelpi-gmail) [reply content/en/v3/admin/guides/jenkins/example.md:124]: The change from a yaml to template provides a better method for merging. Actually I wanted to share what I think the logic is so here goes...

Reviewing the Jenkins chart and how we configured it there are **Chart**, **System** (JenkinsX), and **User** level settings. For plugins management there are _installPlugins_ and _additionalPlugins_ config blocks. The Chart generates only installPlugins and the definiton is:

```
  installPlugins:
    - kubernetes:1.31.3
    - workflow-aggregator:2.6
    - git:4.10.2
    - configuration-as-code:1414.v878271fc496f

```
The System generates only additionalPlugins (inherting the installPlugins from Chart) and the definition is:

```
  additionalPlugins:
  - job-dsl:1.77
  - github-branch-source:2.10.2
  - kubernetes-credentials-provider:0.16
  - jx-resources:1.0.38
  - tekton-client:1.0-alpha3+165
```


If the User needs to add, delete, or update a plugin it will have to be determine whether to include either of the two blocks above. If the requirement is to change or remove an existing plugin you should include the entire block with the change. For example to update the kubernetes plugin and to remove the configuration-as-code plugin you would include the following installPlugins block:

```
  installPlugins:
    - kubernetes:1.31.99                       ## Updated plugin
    - workflow-aggregator:2.6
    - git:4.10.2
#   - configuration-as-code:1414.v878271fc496f ## Removed plugin

```
To add or modify  plugins you would include the following additionalPlugins block:

```
  additionalPlugins:
  - job-dsl:1.79                    ## Modified
  - github-branch-source:2.10.2
  - kubernetes-credentials-provider:0.16
  - jx-resources:1.0.38
  - tekton-client:1.0-alpha3+165
  - jdk-tool:1.5                    ## Added
```

It may be possible to update and "installPlugins" entry by adding it into the "additionalPlugins" list but it has not been tested and it is not recommended.

I can modify this current documentation example to include this above logic. The new version will not have any installPlugins because it will be picked up from the Chart. It will have an enhanced additionalPlugins block which will include plugins for example purposes and the addition for the ingress so that it works. We still have to update verstionStream template to fix the problem but the example doc will still be in good shape. Let me know.

Comment (msvticket) [reply content/en/v3/admin/guides/jenkins/example.md:124]: I'm totally with you that the merging logic you describe is the standard one for helm. (helm/helm#3486 is an old and closed issue about it.)

But you seem to imply that with the help of helmfile it works differently for gotmpl files. Do you have any reference for that?

Comment (tgelpi-gmail) [reply content/en/v3/admin/guides/jenkins/example.md:124]: Don't have a reference. I have been working trying different configurations to see what works. I switched from yaml to template to attempt to have a consistent file naming convention that hopefully that makes the documentation  easy to understand. I thought that using a template might make extending the functionality easier.

Comment (tgelpi-gmail) [reply content/en/v3/admin/guides/jenkins/example.md]: I did some more testing and it looks like you can implement Jenkins in fewer steps than I had previously thought. It can be done with the modification of the values.yaml file only which means that you don't have to have an additional template file (values.yaml.gotmpl) or make additional modifications to the helmfile.yaml. I've updated the document version to reflect this new config. 

In a nutshell you can initalize Jenkins
```
jx gitops jenkins add --name jx-jenkins             ## add Jenkins
jx gitops helmfile resolve --namespace jx-jenkins   ## resolve charts references (optional but helps later on in editing)
```
and update repohome/helmfiles/jx-jenkins/values.yaml with the following:
```
controller:
  ingress:             ## Required versionStream DNS/Ingress break fix
    ingressClassName: nginx
```
This will use all defaults from the Chart and the current version stream template.

If you need to do specific updates to the Jenkins environment you then possibly need to implement additional files and modification to the helmfile.yaml.

**Comment by tgelpi-gmail** [comment]
This version includes some other doc changes provided by others. Looks great!
